spec.html

```diff
@@ -481,40 +676,57 @@ <h1>InitializeInstanceElements ( _O_, _constructor_ )</h1>
-        1. <ins>Let _res_ be ? Call(_element_.[[Start]], _receiver_).</ins>
-        1. <ins>If _res_ is not *undefined*, throw a *TypeError* exception.</ins>
-    1. <ins>For each item _element_ in order from _elements_,</ins>
-      1. <ins>If _element_.[[Placement]] is `"prototype"` or `"static"`, _element_.[[Kind]] is `"hook"`,</ins>
-        1. <ins>If _element_ has a [[Replace]] field,</ins>
-          1. <ins>Assert: _element_.[[Placement]] is `"static"`.</ins>
-          1. <ins>Let _newConstructor_ be Call( _element_.[[Replace]], *undefined*, « _F_ »).</ins>
-          1. <ins>If IsConstructor(_newConstructor_) is *false*, throw a *TypeError* exception.</ins>
-          1. <ins>Set _F_ to _newConstructor_.</ins>
-        1. <ins>If _element_ has a [[Finish]] field,</ins>
-          1. <ins>Let _receiver_ be _F_ if _element_.[[Placement]] is `"static"`, else let _receiver_ be _proto_.</ins>
-          1. <ins>Let _res_ be ? Call(_element_.[[Finish]], _receiver_).</ins>
-          1. <ins>If _res_ is not *undefined*, throw a *TypeError* exception.</ins>
-    1. Return <ins>_F_</ins>.
+        1. Perform ? DefineField(_receiver_, _element_).
+      1. If _element_.[[Placement]] is `"prototype"` or `"static"` and _element_.[[Kind]] is `"hook"`,
+        1. Let _receiver_ be _F_ if _element_.[[Placement]] is `"static"`, else let _receiver_ be _proto_.
+        1. Let _res_ be ? Call(_element_.[[Start]], _receiver_).
+        1. If _res_ is not *undefined*, throw a *TypeError* exception.
+    1. For each item _element_ in order from _elements_,
+      1. If _element_.[[Placement]] is `"prototype"` or `"static"`, _element_.[[Kind]] is `"hook"`,
+        1. If _element_ has a [[Replace]] field,
+          1. Assert: _element_.[[Placement]] is `"static"`.
+          1. Let _newConstructor_ be Call( _element_.[[Replace]], *undefined*, « _F_ »).
+          1. If IsConstructor(_newConstructor_) is *false*, throw a *TypeError* exception.
+          1. Set _F_ to _newConstructor_.
+        1. If _element_ has a [[Finish]] field,
+          1. Let _receiver_ be _F_ if _element_.[[Placement]] is `"static"`, else let _receiver_ be _proto_.
+          1. Let _res_ be ? Call(_element_.[[Finish]], _receiver_).
+          1. If _res_ is not *undefined*, throw a *TypeError* exception.
+    1. Return _F_.
+  </emu-alg>
+  <emu-note type=editor>Brands, methods and accessors are added before initializers so that all methods are visible from all initializers</emu-note>
+</emu-clause>
+
+<emu-clause id="sec-assign-private-names" aoid=AssignPrivateNames>
+  <h1>AssignPrivateNames ( _elements_ )</h1>
+  <emu-alg>
+    1. For each _element_ of _elements_,
+      1. If _element_ has a [[Key]] field and _element_.[[Key]] is a Private Name,
+        1. Let _name_ be _element_.[[Key]].
+        1. If _name_ does not have a [[Type]] field, throw a *TypeError* exception.
```

Comment:
Internal slots are always present, even if undefined, so if `name` is a Private Name here, won't this "if" never fire?

Reply:
This is a Record, not an Object, so it's fine to dynamically add fields (not internal slots).

Reply:
Anyway, the error is here: I should refer to name's Descriptor's Type field